### PR TITLE
[GEN-1026] Output annotation error report

### PR DESCRIPTION
**Pull request by rxu17**
**Purpose:** We want to output the annotation error report from genome nexus annotation as part of the overall goal of providing it to sites to use

**Changes:**
- Added -e flag to annotateMaf in annotation_suite_wrapper.sh to output error report
- Added check_file_existence in annotation_suite_functions.sh to check that the annotation produces both an error report and an annotated file for each input mutation file. Even when there are no failed annotations, there will be a blank error report outputted.
- Note that I formatted the bash scripts using a `shell-formatter` so a lot of changes are due to that

**Testing:**
- For the scope of this ticket, added unit tests for the utility functions used in annotation_suite_wrapper.sh using bats for bash script testing
- manually integration tested by running mutation processing on test pipeline

### CODE REVIEW

File: annotation_suite_wrapper.sh [hunk 1]
```diff
@@ -20,76 +20,90 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
+source ../annotation-tools/annotation_suite_functions.sh
```
Comment (thomasyu888) [inline annotation_suite_wrapper.sh:23]: Will this be an issue within the docker container when triggering this shell script via python code?

Comment (rxu17) [reply annotation_suite_wrapper.sh:23]: Good point, not sure. Will have to try to test this in a docker container

Comment (rxu17) [reply annotation_suite_wrapper.sh:23]: The above doesn't appear to be an issue BUT did discover that our docker container is still running java 8 while the jar file is using java 11. Once this PR is approved, will create a new tag here and then update the java version and tag version for `annotation-tools` in our main genie dockerfile. I don't think we use java anywhere else but cbioportal and annotation-tools

File: README.md
Comment (thomasyu888) [inline README.md:1]: Nice testing!

**Review by thomasyu888 — APPROVED** [review]
🔥 LGTM!  Thanks for putting this together!

**Comment by rxu17** [comment]
@thomasyu888 Added [CONTRIBUTING.md](https://github.com/Sage-Bionetworks/annotation-tools/pull/8/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055), and decided to make the failed annotations error directory [something that is available as an arg to pass in when calling `annotation_suite_wrapper.sh`](https://github.com/Sage-Bionetworks/annotation-tools/pull/8/commits/38229d8bf28c75c7c457eef2df7ca8024087de30) from [process_mutation in main genie repo](https://github.com/Sage-Bionetworks/Genie/pull/544/commits/d7b3bd0a39c3c15fc4f3c870c070519dda509686)

File: annotation_suite_wrapper.sh [hunk 2]
```diff
@@ -20,74 +20,94 @@
...
+    echo "java is required to run this program - aborting..." >&2
+    exit 1
+}
 
 function usage {
     echo "annotation_suite_wrapper.sh"
     echo -e "\t-i | --input-directory               input data directory for processing mutation data files [REQUIRED]"
-    echo -e "\t-o | --output-directory              output directory to write processed and annotated mutation data files to [REQUIRED]"
+    echo -e "\t-o | --output-directory              output directory to write processed and annotated mutation data files [REQUIRED]"
+    echo -e "\t-o | --error-directory               error directory to write annotation error reports to [REQUIRED]"
```
Comment (thomasyu888) [inline annotation_suite_wrapper.sh:39]: Nice addition.  This is just a nice-to-have comment, but wanted to get your thoughts on just having this shell script be written in python. (Not urgent)

**Review by thomasyu888 — APPROVED** [review]
🔥 Nice addition, I agree with your updates

File: annotation_suite_wrapper.sh (hunk 2 above)
Comment (rxu17) [reply annotation_suite_wrapper.sh:39]: Yes I would agree with writing it in python (easier tests, integration with other scripts, etc). Although it depends on if in the near future we still plan to keep using this fork and sync with the gn team's version, or just request for updates to the gn team's version or add wrapper functions to add in our desired changes to their functions?

**Pull request merged**